src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py

```diff
@@ -0,0 +1,390 @@
+        area = wh[..., 0] * wh[..., 1]
+        inter_area = inter[..., 0] * inter[..., 1]
+        iou = inter_area / (area + gt_area - inter_area)
+
+        active_iou = c_iou
+
+        max_iou = _tf.reduce_max(active_iou, 3, keepdims=True)
+        resp_box = _tf.cast(_tf.equal(active_iou, max_iou), dtype=_tf.float32)
+
+        count = _tf.reduce_sum(gt_conf0)
+
+        kr_obj_ij = _tf.stop_gradient(resp_box * gt_conf)
+
+        kr_noobj_ij = 1 - kr_obj_ij
+
+        s = 1 / (self.batch_size * self.grid_shape[0] * self.grid_shape[1])
+
+        kr_obj_ij_plus1 = _tf.expand_dims(kr_obj_ij, -1)
+
+        if rescore:
+            obj_gt_conf = kr_obj_ij * _tf.stop_gradient(iou)
+        else:
+            obj_gt_conf = kr_obj_ij
+
+        kr_box = kr_obj_ij_plus1
+
+        obj_w = (kr_obj_ij * lmb_obj + kr_noobj_ij * lmb_noobj)
+
+        loss_xy = lmb_coord_xy * _tf.reduce_sum(kr_box * _tf.square(gt_xy - xy)) / (count + 0.01)
+
+        loss_wh = _tf.compat.v1.losses.huber_loss (labels=gt_raw_wh, predictions=raw_wh, weights=lmb_coord_wh * kr_box,
+                                                   delta= 1.0)
+
+        # Confidence loss
+        loss_conf = s * _tf.reduce_sum(
+            obj_w * _tf.nn.sigmoid_cross_entropy_with_logits(labels=obj_gt_conf, logits=raw_conf))
+
+        # TODO: tf.nn.softmax_cross_entropy_with_logits_v2 instead of #tf.nn.softmax_cross_entropy_with_logits
+        loss_cls = lmb_class * _tf.reduce_sum(
+            kr_obj_ij * _tf.nn.softmax_cross_entropy_with_logits(labels=gt_class, logits=class_scores)) / (
+                           count + 0.01)
+
+        losses = [loss_xy, loss_wh, loss_conf, loss_cls]
+
+        loss = _tf.add_n(losses)
+        return loss
+
+    def train(self, feed_dict):
+        """
+        Run session for training with new batch of data(Input and Label)
+
+        :param feed_dict: Dictionary
+            Dictionary to store a batch of input data, corresponding labels and iteration number. This is currently
+            passed from the object_detector.py file when a new batch of data is sent.
+        :return: TensorFlow Tensor
+            Loss per batch
+        """
+
+        _, loss_batch = self.sess.run([self.train_op, self.loss], feed_dict={self.images: feed_dict['input'],
+                                                                             self.labels: feed_dict['label']})
+
+        return loss_batch
+
+    def predict(self, feed_dict):
+        """
+        Run session for predicting with new batch of data(Input)
+
+        :param feed_dict: Dictionary
+            Dictionary to store a batch of input data
+        :return: TensorFlow Tensor
+            Feature map from building the network. This will be used in MXNet for further processing
+        """
+
+        output = self.sess.run([self.tf_model], feed_dict={self.images: feed_dict['input']}) #self.labels: feed_dict['label']
```

Comment:
Nit: I think this comment can instead be a TODO, something like "handle labels from validation set"